static/app/views/alerts/rules/metric/insightsMetricField.tsx

```diff
@@ -0,0 +1,221 @@
+import {Fragment, useCallback, useEffect, useMemo} from 'react';
+import styled from '@emotion/styled';
+
+import Tag from 'sentry/components/badge/tag';
+import SelectControl from 'sentry/components/forms/controls/selectControl';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import type {MetricMeta, ParsedMRI} from 'sentry/types/metrics';
+import type {Project} from 'sentry/types/project';
+import {getDefaultAggregation} from 'sentry/utils/metrics';
+import {getReadableMetricType} from 'sentry/utils/metrics/formatters';
+import {
+  DEFAULT_INSIGHTS_METRICS_ALERT_FIELD,
+  DEFAULT_INSIGHTS_MRI,
+  formatMRI,
+  MRIToField,
+  parseField,
+  parseMRI,
+} from 'sentry/utils/metrics/mri';
+import {useVirtualizedMetricsMeta} from 'sentry/utils/metrics/useMetricsMeta';
+import {middleEllipsis} from 'sentry/utils/string/middleEllipsis';
+import {
+  INSIGHTS_METRICS,
+  INSIGHTS_METRICS_OPERATIONS,
+} from 'sentry/views/alerts/rules/metric/utils/isInsightsMetricAlert';
+
+interface Props {
+  aggregate: string;
+  onChange: (value: string, meta: Record<string, any>) => void;
+  project: Project;
+}
+
+// We actually only store a few aggregations for Insights metrics.
+// The `metrics/meta/` endpoint doesn't know this, so hardcode supported aggregations for now.
+const OPERATIONS = [
+  {
+    label: 'avg',
+    value: 'avg',
+  },
+  {
+    label: 'sum',
+    value: 'sum',
+  },
+  {
+    label: 'min',
+    value: 'min',
+  },
+  {
+    label: 'max',
+    value: 'max',
+  },
+  ...INSIGHTS_METRICS_OPERATIONS,
+];
+
+function aggregateRequiresArgs(aggregation?: string) {
+  return aggregation !== 'spm';
+}
+
+function InsightsMetricField({aggregate, project, onChange}: Props) {
+  const {data: meta, isLoading} = useVirtualizedMetricsMeta(
+    {projects: [parseInt(project.id, 10)]},
+    ['spans']
```

Comment:
I see that `spans/webvital.inp`, `spans/mobile.frames_delay` etc are a part of the spans namespace but other webvitals and mobile metrics (ttid/ttfd etc) are still a part of "insights" even though they are on the transactions namespace. Are we planning to include those in the insights alerts workflow too? imo, we should, otherwise it seems arbitrary to the user that they can alert on some metrics here but not others

Reply:
Yup, I'm going to add the other insights metrics from the transaction namespace to the list in another PR! Just need to do some testing on those MRIs